### PR TITLE
Fix links and grammars in various documentations 

### PR DESCRIPTION
**Pull request by garymm**
* Fix lots of links.
* Minor improvements for consistency, clarity or grammar.
* Update jit_python_reference to note the limitations on __exit__.
  (Related to #41420).
* Fix a comment in exit_transforms.cpp: removed the word "not" which
  made the comment say the opposite of the truth.

### CODE REVIEW

**Comment by BowenBao** [comment]
@SplitInfinity do you know anyone from JIT team can take a look?

**Comment by SplitInfinity** [comment]
I can review it

File: torch/csrc/jit/OVERVIEW.md
Comment (SplitInfinity) [inline torch/csrc/jit/OVERVIEW.md:1392]: This should be a lowercase "p" because it is in a file name.

**Review by SplitInfinity — COMMENTED** [review]
Thanks for taking the time to do this!

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md]: Updated to capitalize everywhere.

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md]: I was initially a bit confused because the sentence used both terms ("current thread of execution... invoking task..."). As far as I can tell the terms were being used to refer to a single concept and I found it confusing. After looking into the code a bit, it seems to me that "current thread" and "invoking thread" are both referring to the same thing, while the future may or may be implemented under the hood with a separate thread, so I just reworded it to avoid introducing a new term.

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md:593]: Seems to work in the non-diff Markdown view:
https://github.com/garymm/pytorch/blob/jit-doc/torch/csrc/jit/OVERVIEW.md#conversion-to-ssa

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md]: Done.

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md]: Done.

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md:320]: Precisely. I lengthened it from inc to increment to make it clearer.

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md]: Done, thanks!

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md]: Good point. I went back and tried to make all class names rendered as code for consistency.

File: torch/csrc/jit/OVERVIEW.md
Comment (garymm) [reply torch/csrc/jit/OVERVIEW.md]: I think "is done via methods on Graph objects" is misleading since there is no method on a Graph that can create or delete a Node or Value.  I added another sentence to discuss Block and Value objects. LGTY?

**Comment by SplitInfinity** [comment]
You seem to have force pushed, so I cannot see the changes you made to address my comments. I'm gonna accept since these are just harmless documentation changes.

Thanks once again for doing this!

**Comment by SplitInfinity** [comment]
Actually, can you rebase? I don't think that any of your changes are causing the issue but I cannot merge until those merge conflict errors in the CI system are gone.

**Comment by garymm** [comment]
Rebased.
Sorry about the force push.
Do you prefer authors to use multiple commits during code review?

I am used to amending a single commit to have a clean history.

Thanks for the quick reviews!

**Comment by SplitInfinity** [comment]
> Rebased.
> Sorry about the force push.
> Do you prefer authors to use multiple commits during code review?
> 
> I am used to amending a single commit to have a clean history.
> 
> Thanks for the quick reviews!

Yeah, I prefer multiple commits because it helps me review, especially for the reviews after the first one. We also have tooling that squashes all commits into one before landing, so you don't need to worry about polluting the commit history. The first PR message is used as the commit message.

**Review by SplitInfinity — APPROVED** [review]

**Comment by garymm** [comment]
> We also have tooling that squashes all commits into one before landing, so you don't need to worry about polluting the commit history. 

@SplitInfinity does that happen automatically for all PRs?

**Comment by SplitInfinity** [comment]
> > We also have tooling that squashes all commits into one before landing, so you don't need to worry about polluting the commit history.
> 
> @SplitInfinity does that happen automatically for all PRs?

Yes.

**Comment by garymm** [comment]
I guess all failures are due to flakiness.
Should I file GH issues about the flaky failures?

**Comment by SplitInfinity** [comment]
> I guess all failures are due to flakiness.
> Should I file GH issues about the flaky failures?

No, it's fine. Thanks once again for your help and I welcome more contributions from you in the future!

**Pull request closed without merging**
